distributed/src/test/scala/asobu/distributed/gateway/EndpointSpec.scala

```diff
@@ -0,0 +1,93 @@
+package asobu.distributed.gateway
+
+import akka.actor._
+import asobu.distributed.util.{TestClusterActorSystem}
+import asobu.distributed.{EndpointDefinition, NullaryEndpointDefinition}
+import asobu.distributed.gateway.Endpoint.Prefix
+import asobu.distributed.service.EndpointDefinitionParser
+import org.specs2.mock.Mockito
+import play.api.mvc.RequestHeader
+import play.api.test.{FakeRequest, PlaySpecification}
+import play.core.routing.RouteParams
+
+import play.routes.compiler._
+import shapeless.HNil
+
+object EndpointSpec extends PlaySpecification with Mockito {
+  sequential
+
+  import play.api.http.HttpVerbs._
+
+  val routeString =
+    """
+      |# Some Comments
+      |# Some other Comments
+      |GET   /ep1/a/:n   a.b.c(n: Int)
+      |
+      |GET   /ep2/:a/something   a.b.d(a: Int, b: String)
+      |
+    """.stripMargin
+
+  val createEndpointDef = (route: Route, prefix: Prefix) ⇒ {
+    NullaryEndpointDefinition(prefix, route, ActorPath.fromString("akka://my-sys/user/service-a/worker1"), "role"): EndpointDefinition
+  }
+
+  lazy val parserResult = EndpointDefinitionParser.parse(Prefix("/"), routeString, createEndpointDef)
+  lazy val endPoints = parserResult.right.get
+  lazy val ep1: EndpointDefinition = endPoints(0)
+  lazy val ep2: EndpointDefinition = endPoints(1)
+
+  "Parse to endpoints" >> {
+    parserResult must beRight[List[EndpointDefinition]]
+    parserResult.right.get.length === 2
+
+    "parse comments" >> {
+      val route: Route = ep1.routeInfo
+      route.verb === HttpVerb(GET)
+      route.comments === List(Comment(" Some Comments"), Comment(" Some other Comments"))
+    }
+
+    "parse params" >> {
+      val route: Route = ep1.routeInfo
+      val parameter = route.call.parameters.get.head
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Usage of get on optional type.](https://www.codacy.com/app/iHeartRadio/asobu/file/2322445375/issues/source?bid=3037259&fileBranchId=3178877#l52)